bindings/python/pymongoarrow/pandas_types.py

```diff
@@ -147,6 +147,9 @@ def isna(self):
         )
 
     def __eq__(self, other):
+        # Binary types do not support element-wise comparison.
+        if isinstance(other, Binary):
+            other = np.array(other, dtype=object)
```

Comment:
I think I'm missing something here. Why would a PandasBSONExtensionArray ever compare equal to a bson Binary?

Reply:
I moved to the the subclass

Reply:
I'm still confused. Why does an array need to compare equal to a scalar?

Reply:
Normally it doesn't, but it is part of the test suite that pandas has for extension types.